### PR TITLE
Appeal Status API bug fix and added logging

### PR DESCRIPTION
**Pull request by mc-usds**
Resolves #17976

### Description
Addressing a way that a hash could have possibly been returned instead of a string. Also added logging because this is a case that shouldn't be happening and will need to be investigated further once we have more information.

### Acceptance Criteria
- [ x] Code compiles correctly

### Testing Plan
1. Added a unit test for this edge case.

*Schema Changes Only*

* [ ] Column comments updated
* [ ] Query profiling performed (eyeball Rails log, check bullet and fasterer output)
* [ ] Appropriate indexes added (especially for foreign keys, polymorphic columns, and unique constraints)
* [ ] #appeals-schema notified with summary and link to this PR

### CODE REVIEW

File: spec/models/issue_spec.rb [hunk 1]
```diff
@@ -195,6 +195,11 @@
       let(:codes) { %w[02 15 03 1234] }
       it { is_expected.to be_nil }
     end
+
+    context "when there are more levels than codes" do
+      let(:codes) { %w[01 01] }
+      it { is_expected.to eq("") }
```
Comment (pkarman) [inline spec/models/issue_spec.rb]: could we add a test here that Raven gets called as well? You can see an example of this:

https://github.com/department-of-veterans-affairs/caseflow/blob/master/spec/jobs/decision_issue_sync_job_spec.rb#L54

you'd want to stub `capture_message` instead of `capture_exception` and then check for it as else where in that other spec file.

File: app/models/issue.rb [hunk 2]
```diff
@@ -273,6 +281,7 @@ def friendly_description_for_codes(code_array)
     issue_description
   end
   # rubocop:enable Metrics/CyclomaticComplexity
+  # rubocop:enable Metrics/PerceivedComplexity
```
Comment (pkarman) [inline app/models/issue.rb]: we're trying to avoid adding new exceptions, though in this case I think it's ok since the point is to capture data and then go back and fix this.

File: spec/models/issue_spec.rb (hunk 1 above)
Comment (mc-usds) [reply spec/models/issue_spec.rb]: Added the stub call and the check if it's called.
